subprojects/core/src/main/java/org/gradle/execution/plan/LocalTaskNodeExecutor.java

```diff
@@ -126,10 +127,14 @@ public void visitFileTreeBackedByFile(File file, FileTreeInternal fileTree, File
                         // We would later try to snapshot the inputs anyway, no need to suppress the exception
                         throw e;
                     } else {
-                        validationContext.visitPropertyProblem(
-                            Severity.WARNING,
-                            spec.getPropertyName(),
-                            String.format("cannot be resolved:%n%s%nConsider using Task.dependsOn instead of an input file collection", TextUtil.indent(e.getMessage(), "  "))
+                        validationContext.visitPropertyProblem(problem ->
+                            problem.withId(ValidationProblemId.UNRESOLVABLE_INPUT)
+                                .forProperty(spec.getPropertyName())
+                                .reportAs(Severity.WARNING)
+                                .withDescription(() -> String.format("cannot be resolved:%n%s%n", TextUtil.indent(e.getMessage(), "  ")))
+                                .happensBecause("An input file collection couldn't be resolved, making it impossible to determine task dependencies")
```

Comment:
This is not about the task dependencies. The task dependencies have already been resolved at this point. It makes it impossible to determine the inputs of the task.